# Project 13: Exploring the landscape of the genomic wastewater surveillance ecosystem: a roadmap towards standardisation

## Abstract

Nearly two years after the first report of SARS-CoV-2 in Wuhan, China, the COVID-19 pandemic has affected more than 485 million people. Wastewater surveillance has attracted extensive public attention during the SARS-CoV-2 pandemic, as a passive monitoring system to complement clinical and genomic surveillance activities. Several methods and protocols are already in place that effectively facilitate the detection and quantification of viral RNA in wastewater samples, and concentrations in wastewater have been shown to correlate with trends in reported cases.

With exploratory projects having shown promise, it is now important to coordinate among initiatives for establishing community standards and effectively building an inventory of the available software tools and services, in order to ultimately simplify the deployment of end-to-end genomic wastewater surveillance pipelines and increase the adoption of such promising monitoring methods across the wider community. The first step in this direction would be to identify and catalogue the relevant methodologies and bioinformatics workflows that are integral components of the lifecycle of genomic data derived from wastewater samples, combining into a coherent structure.

The main goal of this project will be to review, collate and offer a first attempt towards integrating, standardising and reporting different approaches that are available for genomic wastewater surveillance. Leveraging the collective expertise of the ELIXIR COVID19 Wastewater Surveillance Working Group, the project will focus on creating a comprehensive landscape of components (e.g. modules, tools etc) that can be effectively utilised for end-to-end genomic wastewater surveillance pipelines. Building on this landscape, this project will attempt a first integration of selected modules within the ELIXIR Tools Platform ecosystem (Galaxy, bio.tools, WorkflowHub etc). Ultimately, this process will be piloted by an effort to integrate and expand new tools into this framework, defining the roadmap towards a standardised  genomic wastewater surveillance ecosystem.



## Topics

Training Platform, Tools Platform
Galaxy Community
COVID-19; Influenza; Emerging Pathogen Monitoring
GA4GH

**Project Number:** 13

### Lead(s)

Ivan Topolsky - ivan.topolsky@sib.swiss
Fotis Psomopoulos - fpsom@certh.gr


## Expected outcomes

Landscape of components (e.g. modules, tools etc) that can be effectively utilised for end-to-end genomic wastewater surveillance pipelines. Examples might include:
modules by the V-pipe teams
tools efforted/supported by other teams joining this project
Implement a collection of genomic wastewater surveillance using the ELIXIR Tools Ecosystem
Create bio.tools “wastewater surveillance” domain
Publishing pipelines on WorkflowHub
Establish a community-based genomic wastewater surveillance methodologies resource, where protocols can be posted and made publicly available.
Pilot the process of adding additional components by integrating them into pipeline(s). This will entail activities such as:
Standardising the formats for interoperability of the involved tools, incl. definitions of variants
A prototype integrated framework around Galaxy
Complementing existing pipelines like V-pipe

check here for correct bullet point formatting

## Expected audience

Researchers active in national wastewater surveillance efforts, experts in genomic data production and management - including metadata standards, bioinformaticians including people who are running wastewater surveillance pipelines
workflow expertise (python, snakemake, nextflow), WES expertise, R/Python/HTML/CSS/Javascript developers
pilot website, UI experience, Data Visualization


**Number of expected hacking days**: 4

