{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "03c0be05-b884-4263-8332-21671f9db5c4",
   "metadata": {},
   "source": [
    "## Set up\n",
    "First you need to install GraphDB locally on you machine\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "abf0b4b2-9f5e-411f-8723-2b55acbf9d0b",
   "metadata": {},
   "outputs": [],
   "source": [
    "from rdflib import ConjunctiveGraph\n",
    "from SPARQLWrapper import SPARQLWrapper, JSON\n",
    "\n",
    "import pandas as pd\n",
    "\n",
    "def remoteQuery(query, endpoint):\n",
    "    endpoint.setQuery(query)\n",
    "    try:\n",
    "        result = endpoint.queryAndConvert()\n",
    "        pd.set_option(\"display.max_rows\",None,\"display.max_colwidth\",5000,\"display.width\",5000,)\n",
    "        df = pd.DataFrame(result['results']['bindings'])\n",
    "        df = df.applymap(lambda x: x['value'])\n",
    "        return df\n",
    "        #return (result['results']['bindings'])\n",
    "    except Exception as e:\n",
    "        print(e)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c3d8bf47-d210-41d7-841f-67fc42549799",
   "metadata": {},
   "source": [
    "\n",
    "### With a GraphDB SPARQL endpoint\n",
    "\n",
    "ep_no_inference points to a SPARQL end point repository in Graphdb where the automatic inferences were deactivated"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "30303cd1-605a-4b2b-89ec-56ce70435175",
   "metadata": {},
   "outputs": [],
   "source": [
    "ep_biotools = SPARQLWrapper(\"http://llamothe-HP-EliteBook-x360-1040-G8-Notebook-PC:7200/repositories/biotools\")\n",
    "ep_biotools.setReturnFormat(JSON)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "281cf006-cc22-4343-b4f9-6113e5699ae1",
   "metadata": {
    "tags": []
   },
   "source": [
    "## query to get top 5 operation reprented in bio.tools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "6fc49cdf-8c4e-4739-a58f-035b97183efc",
   "metadata": {},
   "outputs": [],
   "source": [
    "q= \"\"\"\n",
    "PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>\n",
    "PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>\n",
    "SELECT  ?operation (COUNT(?operation) as ?count) ?label WHERE {\n",
    "?x rdf:type <http://schema.org/SoftwareApplication> ;\n",
    "   <http://schema.org/name> ?name ; \n",
    "   <http://schema.org/featureList> ?operation .\n",
    "   ?operation rdfs:label ?label . \n",
    "} GROUP BY ?operation ?label\n",
    "ORDER BY DESC(?count)\n",
    "LIMIT 5\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "45077694-a648-43fb-a1ed-f958538edae4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "nb data/operation with has_topic property: 5\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>operation</th>\n",
       "      <th>count</th>\n",
       "      <th>label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>http://edamontology.org/operation_0337</td>\n",
       "      <td>2075</td>\n",
       "      <td>Visualisation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>http://edamontology.org/operation_3435</td>\n",
       "      <td>1581</td>\n",
       "      <td>Standardisation and normalisation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>http://edamontology.org/operation_3196</td>\n",
       "      <td>1351</td>\n",
       "      <td>Genotyping</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>http://edamontology.org/operation_2422</td>\n",
       "      <td>1340</td>\n",
       "      <td>Data retrieval</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>http://edamontology.org/operation_2495</td>\n",
       "      <td>1139</td>\n",
       "      <td>Expression analysis</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                operation count                              label\n",
       "0  http://edamontology.org/operation_0337  2075                      Visualisation\n",
       "1  http://edamontology.org/operation_3435  1581  Standardisation and normalisation\n",
       "2  http://edamontology.org/operation_3196  1351                         Genotyping\n",
       "3  http://edamontology.org/operation_2422  1340                     Data retrieval\n",
       "4  http://edamontology.org/operation_2495  1139                Expression analysis"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(f\"nb data/operation with has_topic property: {len(remoteQuery(query=q, endpoint=ep_biotools))}\")\n",
    "remoteQuery(query=q, endpoint=ep_biotools)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e4206c5a-24ef-49af-a88b-3e2cee8ce960",
   "metadata": {
    "tags": []
   },
   "source": [
    "## query to get the top 100 tools and their associated EDAM def and synonyms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "3a9d11e7-0546-4e92-8dd8-616028fe8fd5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "nb data/operation with has_topic property: 100\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>biotools_id</th>\n",
       "      <th>name</th>\n",
       "      <th>feature</th>\n",
       "      <th>def</th>\n",
       "      <th>syn</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>https://bio.tools/-CNN</td>\n",
       "      <td>-CNN</td>\n",
       "      <td>http://edamontology.org/operation_3215</td>\n",
       "      <td>Identify peaks in a spectrum from a mass spectrometry, NMR, or some other spectrum-generating experiment.</td>\n",
       "      <td>Peak assignment</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>https://bio.tools/-CNN</td>\n",
       "      <td>-CNN</td>\n",
       "      <td>http://edamontology.org/operation_3215</td>\n",
       "      <td>Identify peaks in a spectrum from a mass spectrometry, NMR, or some other spectrum-generating experiment.</td>\n",
       "      <td>Peak finding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>https://bio.tools/-CNN</td>\n",
       "      <td>-CNN</td>\n",
       "      <td>http://edamontology.org/operation_3222</td>\n",
       "      <td>Identify putative protein-binding regions in a genome sequence from analysis of Chip-sequencing data or ChIP-on-chip data.</td>\n",
       "      <td>Protein binding peak detection</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>https://bio.tools/-TRIS</td>\n",
       "      <td>-TRIS</td>\n",
       "      <td>http://edamontology.org/operation_2422</td>\n",
       "      <td>Retrieve an entry (or part of an entry) from a data resource that matches a supplied query. This might include some primary data and annotation. The query is a data identifier or other indexed term. For example, retrieve a sequence record with the specified accession number, or matching supplied keywords.</td>\n",
       "      <td>Data extraction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>https://bio.tools/-TRIS</td>\n",
       "      <td>-TRIS</td>\n",
       "      <td>http://edamontology.org/operation_2422</td>\n",
       "      <td>Retrieve an entry (or part of an entry) from a data resource that matches a supplied query. This might include some primary data and annotation. The query is a data identifier or other indexed term. For example, retrieve a sequence record with the specified accession number, or matching supplied keywords.</td>\n",
       "      <td>Retrieval</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>https://bio.tools/-TRIS</td>\n",
       "      <td>-TRIS</td>\n",
       "      <td>http://edamontology.org/operation_3198</td>\n",
       "      <td>Align short oligonucleotide sequences (reads) to a larger (genomic) sequence.</td>\n",
       "      <td>Oligonucleotide alignment</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>https://bio.tools/-TRIS</td>\n",
       "      <td>-TRIS</td>\n",
       "      <td>http://edamontology.org/operation_3198</td>\n",
       "      <td>Align short oligonucleotide sequences (reads) to a larger (genomic) sequence.</td>\n",
       "      <td>Oligonucleotide alignment construction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>https://bio.tools/-TRIS</td>\n",
       "      <td>-TRIS</td>\n",
       "      <td>http://edamontology.org/operation_3198</td>\n",
       "      <td>Align short oligonucleotide sequences (reads) to a larger (genomic) sequence.</td>\n",
       "      <td>Oligonucleotide alignment generation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>https://bio.tools/-TRIS</td>\n",
       "      <td>-TRIS</td>\n",
       "      <td>http://edamontology.org/operation_3198</td>\n",
       "      <td>Align short oligonucleotide sequences (reads) to a larger (genomic) sequence.</td>\n",
       "      <td>Oligonucleotide mapping</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>https://bio.tools/-TRIS</td>\n",
       "      <td>-TRIS</td>\n",
       "      <td>http://edamontology.org/operation_3198</td>\n",
       "      <td>Align short oligonucleotide sequences (reads) to a larger (genomic) sequence.</td>\n",
       "      <td>Read alignment</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>https://bio.tools/-TRIS</td>\n",
       "      <td>-TRIS</td>\n",
       "      <td>http://edamontology.org/operation_3198</td>\n",
       "      <td>Align short oligonucleotide sequences (reads) to a larger (genomic) sequence.</td>\n",
       "      <td>Short oligonucleotide alignment</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>https://bio.tools/-TRIS</td>\n",
       "      <td>-TRIS</td>\n",
       "      <td>http://edamontology.org/operation_3198</td>\n",
       "      <td>Align short oligonucleotide sequences (reads) to a larger (genomic) sequence.</td>\n",
       "      <td>Short read alignment</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>https://bio.tools/-TRIS</td>\n",
       "      <td>-TRIS</td>\n",
       "      <td>http://edamontology.org/operation_3198</td>\n",
       "      <td>Align short oligonucleotide sequences (reads) to a larger (genomic) sequence.</td>\n",
       "      <td>Short read mapping</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>https://bio.tools/-TRIS</td>\n",
       "      <td>-TRIS</td>\n",
       "      <td>http://edamontology.org/operation_3198</td>\n",
       "      <td>Align short oligonucleotide sequences (reads) to a larger (genomic) sequence.</td>\n",
       "      <td>Short sequence read mapping</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>https://bio.tools/-TRIS</td>\n",
       "      <td>-TRIS</td>\n",
       "      <td>http://edamontology.org/operation_3799</td>\n",
       "      <td>Counting and measuring experimentally determined observations into quantities.</td>\n",
       "      <td>Quantitation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>https://bio.tools/1000genomes</td>\n",
       "      <td>1000Genomes</td>\n",
       "      <td>http://edamontology.org/operation_0306</td>\n",
       "      <td>Process and analyse text (typically scientific literature) to extract information from it.</td>\n",
       "      <td>Literature mining</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>https://bio.tools/1000genomes</td>\n",
       "      <td>1000Genomes</td>\n",
       "      <td>http://edamontology.org/operation_0306</td>\n",
       "      <td>Process and analyse text (typically scientific literature) to extract information from it.</td>\n",
       "      <td>Text analytics</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>https://bio.tools/1000genomes</td>\n",
       "      <td>1000Genomes</td>\n",
       "      <td>http://edamontology.org/operation_0306</td>\n",
       "      <td>Process and analyse text (typically scientific literature) to extract information from it.</td>\n",
       "      <td>Text data mining</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>https://bio.tools/1000genomes</td>\n",
       "      <td>1000Genomes</td>\n",
       "      <td>http://edamontology.org/operation_3431</td>\n",
       "      <td>Deposit some data in a database or some other type of repository or software system.</td>\n",
       "      <td>Data deposition</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>https://bio.tools/1000genomes</td>\n",
       "      <td>1000Genomes</td>\n",
       "      <td>http://edamontology.org/operation_3431</td>\n",
       "      <td>Deposit some data in a database or some other type of repository or software system.</td>\n",
       "      <td>Data submission</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>https://bio.tools/1000genomes</td>\n",
       "      <td>1000Genomes</td>\n",
       "      <td>http://edamontology.org/operation_3431</td>\n",
       "      <td>Deposit some data in a database or some other type of repository or software system.</td>\n",
       "      <td>Database deposition</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>https://bio.tools/1000genomes</td>\n",
       "      <td>1000Genomes</td>\n",
       "      <td>http://edamontology.org/operation_3431</td>\n",
       "      <td>Deposit some data in a database or some other type of repository or software system.</td>\n",
       "      <td>Database submission</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>https://bio.tools/1000genomes</td>\n",
       "      <td>1000Genomes</td>\n",
       "      <td>http://edamontology.org/operation_3431</td>\n",
       "      <td>Deposit some data in a database or some other type of repository or software system.</td>\n",
       "      <td>Submission</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>https://bio.tools/1000genomes_assembly_converter</td>\n",
       "      <td>1000Genomes assembly converter</td>\n",
       "      <td>http://edamontology.org/operation_2944</td>\n",
       "      <td>Generate a physical (sequence) map of a DNA sequence showing the physical distance (base pairs) between features or landmarks such as restriction sites, cloned DNA fragments, genes and other genetic markers.</td>\n",
       "      <td>Physical cartography</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>https://bio.tools/1000genomes_id_history_converter</td>\n",
       "      <td>1000Genomes ID history converter</td>\n",
       "      <td>http://edamontology.org/operation_3282</td>\n",
       "      <td>Map data identifiers to one another for example to establish a link between two biological databases for the purposes of data integration.</td>\n",
       "      <td>Accession mapping</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>https://bio.tools/1000genomes_id_history_converter</td>\n",
       "      <td>1000Genomes ID history converter</td>\n",
       "      <td>http://edamontology.org/operation_3282</td>\n",
       "      <td>Map data identifiers to one another for example to establish a link between two biological databases for the purposes of data integration.</td>\n",
       "      <td>Identifier mapping</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>https://bio.tools/1000genomes_vcf2ped</td>\n",
       "      <td>1000Genomes VCF2PED</td>\n",
       "      <td>http://edamontology.org/operation_0335</td>\n",
       "      <td>Reformat a file of data (or equivalent entity in memory).</td>\n",
       "      <td>File format conversion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>https://bio.tools/1000genomes_vcf2ped</td>\n",
       "      <td>1000Genomes VCF2PED</td>\n",
       "      <td>http://edamontology.org/operation_0335</td>\n",
       "      <td>Reformat a file of data (or equivalent entity in memory).</td>\n",
       "      <td>File formatting</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>https://bio.tools/1000genomes_vcf2ped</td>\n",
       "      <td>1000Genomes VCF2PED</td>\n",
       "      <td>http://edamontology.org/operation_0335</td>\n",
       "      <td>Reformat a file of data (or equivalent entity in memory).</td>\n",
       "      <td>File reformatting</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>https://bio.tools/1000genomes_vcf2ped</td>\n",
       "      <td>1000Genomes VCF2PED</td>\n",
       "      <td>http://edamontology.org/operation_0335</td>\n",
       "      <td>Reformat a file of data (or equivalent entity in memory).</td>\n",
       "      <td>Format conversion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>https://bio.tools/1000genomes_vcf2ped</td>\n",
       "      <td>1000Genomes VCF2PED</td>\n",
       "      <td>http://edamontology.org/operation_0335</td>\n",
       "      <td>Reformat a file of data (or equivalent entity in memory).</td>\n",
       "      <td>Reformatting</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>https://bio.tools/13Check_RNA</td>\n",
       "      <td>13Check_RNA</td>\n",
       "      <td>http://edamontology.org/operation_3431</td>\n",
       "      <td>Deposit some data in a database or some other type of repository or software system.</td>\n",
       "      <td>Data deposition</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>https://bio.tools/13Check_RNA</td>\n",
       "      <td>13Check_RNA</td>\n",
       "      <td>http://edamontology.org/operation_3431</td>\n",
       "      <td>Deposit some data in a database or some other type of repository or software system.</td>\n",
       "      <td>Data submission</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>https://bio.tools/13Check_RNA</td>\n",
       "      <td>13Check_RNA</td>\n",
       "      <td>http://edamontology.org/operation_3431</td>\n",
       "      <td>Deposit some data in a database or some other type of repository or software system.</td>\n",
       "      <td>Database deposition</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>https://bio.tools/13Check_RNA</td>\n",
       "      <td>13Check_RNA</td>\n",
       "      <td>http://edamontology.org/operation_3431</td>\n",
       "      <td>Deposit some data in a database or some other type of repository or software system.</td>\n",
       "      <td>Database submission</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>https://bio.tools/13Check_RNA</td>\n",
       "      <td>13Check_RNA</td>\n",
       "      <td>http://edamontology.org/operation_3431</td>\n",
       "      <td>Deposit some data in a database or some other type of repository or software system.</td>\n",
       "      <td>Submission</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>https://bio.tools/1433pred</td>\n",
       "      <td>14-3-3-Pred</td>\n",
       "      <td>http://edamontology.org/operation_2479</td>\n",
       "      <td>Analyse a protein sequence (using methods that are only applicable to protein sequences).</td>\n",
       "      <td>Sequence analysis (protein)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>https://bio.tools/1433pred</td>\n",
       "      <td>14-3-3-Pred</td>\n",
       "      <td>http://edamontology.org/operation_2575</td>\n",
       "      <td>Identify or predict catalytic residues, active sites or other ligand-binding sites in protein sequences or structures.</td>\n",
       "      <td>Protein binding site detection</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>https://bio.tools/1433pred</td>\n",
       "      <td>14-3-3-Pred</td>\n",
       "      <td>http://edamontology.org/operation_2575</td>\n",
       "      <td>Identify or predict catalytic residues, active sites or other ligand-binding sites in protein sequences or structures.</td>\n",
       "      <td>Protein binding site prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>https://bio.tools/16S</td>\n",
       "      <td>16S</td>\n",
       "      <td>http://edamontology.org/operation_3799</td>\n",
       "      <td>Counting and measuring experimentally determined observations into quantities.</td>\n",
       "      <td>Quantitation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>https://bio.tools/16S</td>\n",
       "      <td>16S</td>\n",
       "      <td>http://edamontology.org/operation_3192</td>\n",
       "      <td>Cut (remove) the end from a molecular sequence.</td>\n",
       "      <td>Trimming</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>https://bio.tools/16s_classifier</td>\n",
       "      <td>16S classifier</td>\n",
       "      <td>http://edamontology.org/operation_3460</td>\n",
       "      <td>Classifiication (typically of molecular sequences) by assignment to some taxonomic hierarchy.</td>\n",
       "      <td>Taxonomy assignment</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>https://bio.tools/16spip</td>\n",
       "      <td>16sPIP</td>\n",
       "      <td>http://edamontology.org/operation_2454</td>\n",
       "      <td>Detect, predict and identify genes or components of genes in DNA sequences, including promoters, coding regions, splice sites, etc.</td>\n",
       "      <td>Gene calling</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>https://bio.tools/16spip</td>\n",
       "      <td>16sPIP</td>\n",
       "      <td>http://edamontology.org/operation_2454</td>\n",
       "      <td>Detect, predict and identify genes or components of genes in DNA sequences, including promoters, coding regions, splice sites, etc.</td>\n",
       "      <td>Gene finding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>https://bio.tools/19andme</td>\n",
       "      <td>19andMe</td>\n",
       "      <td>http://edamontology.org/operation_3283</td>\n",
       "      <td>Process data in such a way that makes it hard to trace to the person which the data concerns.</td>\n",
       "      <td>Data anonymisation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>https://bio.tools/1KP</td>\n",
       "      <td>1KP</td>\n",
       "      <td>http://edamontology.org/operation_3891</td>\n",
       "      <td>Compute Essential Dynamics (ED) on a simulation trajectory: an analysis of molecule dynamics using PCA (Principal Component Analysis) applied to the atomic positional fluctuations.</td>\n",
       "      <td>PCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>https://bio.tools/1KP</td>\n",
       "      <td>1KP</td>\n",
       "      <td>http://edamontology.org/operation_3891</td>\n",
       "      <td>Compute Essential Dynamics (ED) on a simulation trajectory: an analysis of molecule dynamics using PCA (Principal Component Analysis) applied to the atomic positional fluctuations.</td>\n",
       "      <td>ED</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>https://bio.tools/1KP</td>\n",
       "      <td>1KP</td>\n",
       "      <td>http://edamontology.org/operation_3891</td>\n",
       "      <td>Compute Essential Dynamics (ED) on a simulation trajectory: an analysis of molecule dynamics using PCA (Principal Component Analysis) applied to the atomic positional fluctuations.</td>\n",
       "      <td>Principal modes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>https://bio.tools/1d-dft-geometrical</td>\n",
       "      <td>Geometrical interpretation of the 1-D DFT</td>\n",
       "      <td>http://edamontology.org/operation_0337</td>\n",
       "      <td>Visualise, plot or render (graphically) biomolecular data such as molecular sequences or structures.</td>\n",
       "      <td>Data visualisation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>https://bio.tools/1d-dft-geometrical</td>\n",
       "      <td>Geometrical interpretation of the 1-D DFT</td>\n",
       "      <td>http://edamontology.org/operation_0337</td>\n",
       "      <td>Visualise, plot or render (graphically) biomolecular data such as molecular sequences or structures.</td>\n",
       "      <td>Rendering</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>https://bio.tools/2-kupl</td>\n",
       "      <td>2-kupl</td>\n",
       "      <td>http://edamontology.org/operation_3192</td>\n",
       "      <td>Cut (remove) the end from a molecular sequence.</td>\n",
       "      <td>Trimming</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>https://bio.tools/2-kupl</td>\n",
       "      <td>2-kupl</td>\n",
       "      <td>http://edamontology.org/operation_0524</td>\n",
       "      <td>Sequence assembly by combining fragments without the aid of a reference sequence or genome.</td>\n",
       "      <td>De Bruijn graph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>https://bio.tools/2-kupl</td>\n",
       "      <td>2-kupl</td>\n",
       "      <td>http://edamontology.org/operation_0524</td>\n",
       "      <td>Sequence assembly by combining fragments without the aid of a reference sequence or genome.</td>\n",
       "      <td>Sequence assembly (de-novo assembly)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>https://bio.tools/2-kupl</td>\n",
       "      <td>2-kupl</td>\n",
       "      <td>http://edamontology.org/operation_3227</td>\n",
       "      <td>Detect, identify and map mutations, such as single nucleotide polymorphisms, short indels and structural variants, in multiple DNA sequences. Typically the alignment and comparison of the fluorescent traces produced by DNA sequencing hardware, to study genomic alterations.</td>\n",
       "      <td>Variant mapping</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>https://bio.tools/2018</td>\n",
       "      <td>2018</td>\n",
       "      <td>http://edamontology.org/operation_3192</td>\n",
       "      <td>Cut (remove) the end from a molecular sequence.</td>\n",
       "      <td>Trimming</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>https://bio.tools/2018_YPIC_Challenge</td>\n",
       "      <td>2018 YPIC Challenge</td>\n",
       "      <td>http://edamontology.org/operation_0417</td>\n",
       "      <td>Predict post-translation modification sites in protein sequences.</td>\n",
       "      <td>PTM analysis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>https://bio.tools/2018_YPIC_Challenge</td>\n",
       "      <td>2018 YPIC Challenge</td>\n",
       "      <td>http://edamontology.org/operation_0417</td>\n",
       "      <td>Predict post-translation modification sites in protein sequences.</td>\n",
       "      <td>PTM prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>https://bio.tools/2018_YPIC_Challenge</td>\n",
       "      <td>2018 YPIC Challenge</td>\n",
       "      <td>http://edamontology.org/operation_0417</td>\n",
       "      <td>Predict post-translation modification sites in protein sequences.</td>\n",
       "      <td>PTM site analysis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>https://bio.tools/2018_YPIC_Challenge</td>\n",
       "      <td>2018 YPIC Challenge</td>\n",
       "      <td>http://edamontology.org/operation_0417</td>\n",
       "      <td>Predict post-translation modification sites in protein sequences.</td>\n",
       "      <td>Post-translation modification site prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>https://bio.tools/2018_YPIC_Challenge</td>\n",
       "      <td>2018 YPIC Challenge</td>\n",
       "      <td>http://edamontology.org/operation_0417</td>\n",
       "      <td>Predict post-translation modification sites in protein sequences.</td>\n",
       "      <td>Post-translational modification analysis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>https://bio.tools/2018_YPIC_Challenge</td>\n",
       "      <td>2018 YPIC Challenge</td>\n",
       "      <td>http://edamontology.org/operation_0417</td>\n",
       "      <td>Predict post-translation modification sites in protein sequences.</td>\n",
       "      <td>PTM site prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>https://bio.tools/2018_YPIC_Challenge</td>\n",
       "      <td>2018 YPIC Challenge</td>\n",
       "      <td>http://edamontology.org/operation_0417</td>\n",
       "      <td>Predict post-translation modification sites in protein sequences.</td>\n",
       "      <td>Protein post-translation modification site prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>https://bio.tools/2018_YPIC_Challenge</td>\n",
       "      <td>2018 YPIC Challenge</td>\n",
       "      <td>http://edamontology.org/operation_0476</td>\n",
       "      <td>Predict tertiary structure of protein sequence(s) without homologs of known structure.</td>\n",
       "      <td>de novo structure prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>https://bio.tools/2019ncovas</td>\n",
       "      <td>2019nCoVAS</td>\n",
       "      <td>http://edamontology.org/operation_0324</td>\n",
       "      <td>Analyse an existing phylogenetic tree or trees, typically to detect features or make predictions.</td>\n",
       "      <td>Phylogenetic tree analysis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>https://bio.tools/2019ncovas</td>\n",
       "      <td>2019nCoVAS</td>\n",
       "      <td>http://edamontology.org/operation_2403</td>\n",
       "      <td>Analyse one or more known molecular sequences.</td>\n",
       "      <td>Sequence analysis (general)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>https://bio.tools/2020plus</td>\n",
       "      <td>2020plus</td>\n",
       "      <td>http://edamontology.org/operation_2454</td>\n",
       "      <td>Detect, predict and identify genes or components of genes in DNA sequences, including promoters, coding regions, splice sites, etc.</td>\n",
       "      <td>Gene calling</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>https://bio.tools/2020plus</td>\n",
       "      <td>2020plus</td>\n",
       "      <td>http://edamontology.org/operation_2454</td>\n",
       "      <td>Detect, predict and identify genes or components of genes in DNA sequences, including promoters, coding regions, splice sites, etc.</td>\n",
       "      <td>Gene finding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>https://bio.tools/2020plus</td>\n",
       "      <td>2020plus</td>\n",
       "      <td>http://edamontology.org/operation_3197</td>\n",
       "      <td>Analyse a genetic variation, for example to annotate its location, alleles, classification, and effects on individual transcripts predicted for a gene model.</td>\n",
       "      <td>Genetic variation annotation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>https://bio.tools/2020plus</td>\n",
       "      <td>2020plus</td>\n",
       "      <td>http://edamontology.org/operation_3197</td>\n",
       "      <td>Analyse a genetic variation, for example to annotate its location, alleles, classification, and effects on individual transcripts predicted for a gene model.</td>\n",
       "      <td>Sequence variation analysis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>https://bio.tools/2020plus</td>\n",
       "      <td>2020plus</td>\n",
       "      <td>http://edamontology.org/operation_3197</td>\n",
       "      <td>Analyse a genetic variation, for example to annotate its location, alleles, classification, and effects on individual transcripts predicted for a gene model.</td>\n",
       "      <td>Variant analysis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>https://bio.tools/2DProt</td>\n",
       "      <td>2DProts</td>\n",
       "      <td>http://edamontology.org/operation_2486</td>\n",
       "      <td>Render a topology diagram of protein secondary structure.</td>\n",
       "      <td>Topology diagram rendering</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>https://bio.tools/2StrucCompare</td>\n",
       "      <td>2StrucCompare</td>\n",
       "      <td>http://edamontology.org/operation_0267</td>\n",
       "      <td>Predict secondary structure of protein sequences.</td>\n",
       "      <td>Secondary structure prediction (protein)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>https://bio.tools/2StrucCompare</td>\n",
       "      <td>2StrucCompare</td>\n",
       "      <td>http://edamontology.org/operation_2488</td>\n",
       "      <td>Compare protein secondary structures.</td>\n",
       "      <td>Protein secondary structure</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>https://bio.tools/2StrucCompare</td>\n",
       "      <td>2StrucCompare</td>\n",
       "      <td>http://edamontology.org/operation_2488</td>\n",
       "      <td>Compare protein secondary structures.</td>\n",
       "      <td>Secondary structure comparison (protein)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>https://bio.tools/2d-image-cepstral-analysis</td>\n",
       "      <td>2D Image cepstral analysis</td>\n",
       "      <td>http://edamontology.org/operation_3860</td>\n",
       "      <td>Calculate a theoretical mass spectrometry spectra for given sequences.</td>\n",
       "      <td>Spectrum prediction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>https://bio.tools/2d-mh</td>\n",
       "      <td>2D-MH</td>\n",
       "      <td>http://edamontology.org/operation_2479</td>\n",
       "      <td>Analyse a protein sequence (using methods that are only applicable to protein sequences).</td>\n",
       "      <td>Sequence analysis (protein)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>https://bio.tools/2d-mh</td>\n",
       "      <td>2D-MH</td>\n",
       "      <td>http://edamontology.org/operation_0337</td>\n",
       "      <td>Visualise, plot or render (graphically) biomolecular data such as molecular sequences or structures.</td>\n",
       "      <td>Data visualisation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>https://bio.tools/2d-mh</td>\n",
       "      <td>2D-MH</td>\n",
       "      <td>http://edamontology.org/operation_0337</td>\n",
       "      <td>Visualise, plot or render (graphically) biomolecular data such as molecular sequences or structures.</td>\n",
       "      <td>Rendering</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>https://bio.tools/2d-page</td>\n",
       "      <td>2D-PAGE</td>\n",
       "      <td>http://edamontology.org/operation_0224</td>\n",
       "      <td>Search or query a data resource and retrieve entries and / or annotation.</td>\n",
       "      <td>Database retrieval</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>https://bio.tools/2dSS</td>\n",
       "      <td>2dSS</td>\n",
       "      <td>http://edamontology.org/operation_0570</td>\n",
       "      <td>Visualise or render molecular 3D structure, for example a high-quality static picture or animation.</td>\n",
       "      <td>Structure rendering</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>https://bio.tools/2de-pattern</td>\n",
       "      <td>2DE-pattern</td>\n",
       "      <td>http://edamontology.org/operation_2421</td>\n",
       "      <td>Search a database (or other data resource) with a supplied query and retrieve entries (or parts of entries) that are similar to the query.</td>\n",
       "      <td>Search</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81</th>\n",
       "      <td>https://bio.tools/2dimpute</td>\n",
       "      <td>2DImpute</td>\n",
       "      <td>http://edamontology.org/operation_3557</td>\n",
       "      <td>Replace missing data with substituted values, usually by using some statistical or other mathematical approach.</td>\n",
       "      <td>Data imputation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>https://bio.tools/2dkd</td>\n",
       "      <td>2DKD</td>\n",
       "      <td>http://edamontology.org/operation_2422</td>\n",
       "      <td>Retrieve an entry (or part of an entry) from a data resource that matches a supplied query. This might include some primary data and annotation. The query is a data identifier or other indexed term. For example, retrieve a sequence record with the specified accession number, or matching supplied keywords.</td>\n",
       "      <td>Data extraction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>https://bio.tools/2dkd</td>\n",
       "      <td>2DKD</td>\n",
       "      <td>http://edamontology.org/operation_2422</td>\n",
       "      <td>Retrieve an entry (or part of an entry) from a data resource that matches a supplied query. This might include some primary data and annotation. The query is a data identifier or other indexed term. For example, retrieve a sequence record with the specified accession number, or matching supplied keywords.</td>\n",
       "      <td>Retrieval</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>https://bio.tools/2dkd</td>\n",
       "      <td>2DKD</td>\n",
       "      <td>http://edamontology.org/operation_2421</td>\n",
       "      <td>Search a database (or other data resource) with a supplied query and retrieve entries (or parts of entries) that are similar to the query.</td>\n",
       "      <td>Search</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>https://bio.tools/2kplus2</td>\n",
       "      <td>2kplus2</td>\n",
       "      <td>http://edamontology.org/operation_2421</td>\n",
       "      <td>Search a database (or other data resource) with a supplied query and retrieve entries (or parts of entries) that are similar to the query.</td>\n",
       "      <td>Search</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>https://bio.tools/2kplus2</td>\n",
       "      <td>2kplus2</td>\n",
       "      <td>http://edamontology.org/operation_0484</td>\n",
       "      <td>Find single nucleotide polymorphisms (SNPs) - single nucleotide change in base positions - between sequences. Typically done for sequences from a high-throughput sequencing experiment that differ from a reference genome and which might, especially by reference to population frequency or functional data, indicate a polymorphism.</td>\n",
       "      <td>SNP calling</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>https://bio.tools/2kplus2</td>\n",
       "      <td>2kplus2</td>\n",
       "      <td>http://edamontology.org/operation_0484</td>\n",
       "      <td>Find single nucleotide polymorphisms (SNPs) - single nucleotide change in base positions - between sequences. Typically done for sequences from a high-throughput sequencing experiment that differ from a reference genome and which might, especially by reference to population frequency or functional data, indicate a polymorphism.</td>\n",
       "      <td>SNP discovery</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>https://bio.tools/2kplus2</td>\n",
       "      <td>2kplus2</td>\n",
       "      <td>http://edamontology.org/operation_0484</td>\n",
       "      <td>Find single nucleotide polymorphisms (SNPs) - single nucleotide change in base positions - between sequences. Typically done for sequences from a high-throughput sequencing experiment that differ from a reference genome and which might, especially by reference to population frequency or functional data, indicate a polymorphism.</td>\n",
       "      <td>Single nucleotide polymorphism detection</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>https://bio.tools/2lpirnapred</td>\n",
       "      <td>2lpiRNApred</td>\n",
       "      <td>http://edamontology.org/operation_3936</td>\n",
       "      <td>A dimensionality reduction process that selects a subset of relevant features (variables, predictors) for use in model construction.</td>\n",
       "      <td>Attribute selection</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>https://bio.tools/2lpirnapred</td>\n",
       "      <td>2lpiRNApred</td>\n",
       "      <td>http://edamontology.org/operation_3936</td>\n",
       "      <td>A dimensionality reduction process that selects a subset of relevant features (variables, predictors) for use in model construction.</td>\n",
       "      <td>Variable selection</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>https://bio.tools/2lpirnapred</td>\n",
       "      <td>2lpiRNApred</td>\n",
       "      <td>http://edamontology.org/operation_3936</td>\n",
       "      <td>A dimensionality reduction process that selects a subset of relevant features (variables, predictors) for use in model construction.</td>\n",
       "      <td>Variable subset selection</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>https://bio.tools/2lpirnapred</td>\n",
       "      <td>2lpiRNApred</td>\n",
       "      <td>http://edamontology.org/operation_3937</td>\n",
       "      <td>A dimensionality reduction process which builds (ideally) informative and non-redundant values (features) from an initial set of measured data, to aid subsequent generalization, learning or interpretation.</td>\n",
       "      <td>Feature projection</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>https://bio.tools/2passtools</td>\n",
       "      <td>2passtools</td>\n",
       "      <td>http://edamontology.org/operation_3198</td>\n",
       "      <td>Align short oligonucleotide sequences (reads) to a larger (genomic) sequence.</td>\n",
       "      <td>Oligonucleotide alignment</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>https://bio.tools/2passtools</td>\n",
       "      <td>2passtools</td>\n",
       "      <td>http://edamontology.org/operation_3198</td>\n",
       "      <td>Align short oligonucleotide sequences (reads) to a larger (genomic) sequence.</td>\n",
       "      <td>Oligonucleotide alignment construction</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>https://bio.tools/2passtools</td>\n",
       "      <td>2passtools</td>\n",
       "      <td>http://edamontology.org/operation_3198</td>\n",
       "      <td>Align short oligonucleotide sequences (reads) to a larger (genomic) sequence.</td>\n",
       "      <td>Oligonucleotide alignment generation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>https://bio.tools/2passtools</td>\n",
       "      <td>2passtools</td>\n",
       "      <td>http://edamontology.org/operation_3198</td>\n",
       "      <td>Align short oligonucleotide sequences (reads) to a larger (genomic) sequence.</td>\n",
       "      <td>Oligonucleotide mapping</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>https://bio.tools/2passtools</td>\n",
       "      <td>2passtools</td>\n",
       "      <td>http://edamontology.org/operation_3198</td>\n",
       "      <td>Align short oligonucleotide sequences (reads) to a larger (genomic) sequence.</td>\n",
       "      <td>Read alignment</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>https://bio.tools/2passtools</td>\n",
       "      <td>2passtools</td>\n",
       "      <td>http://edamontology.org/operation_3198</td>\n",
       "      <td>Align short oligonucleotide sequences (reads) to a larger (genomic) sequence.</td>\n",
       "      <td>Short oligonucleotide alignment</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>https://bio.tools/2passtools</td>\n",
       "      <td>2passtools</td>\n",
       "      <td>http://edamontology.org/operation_3198</td>\n",
       "      <td>Align short oligonucleotide sequences (reads) to a larger (genomic) sequence.</td>\n",
       "      <td>Short read alignment</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           biotools_id                                       name                                 feature                                                                                                                                                                                                                                                                                                                                        def                                                    syn\n",
       "0                               https://bio.tools/-CNN                                       -CNN  http://edamontology.org/operation_3215                                                                                                                                                                                                                                  Identify peaks in a spectrum from a mass spectrometry, NMR, or some other spectrum-generating experiment.                                        Peak assignment\n",
       "1                               https://bio.tools/-CNN                                       -CNN  http://edamontology.org/operation_3215                                                                                                                                                                                                                                  Identify peaks in a spectrum from a mass spectrometry, NMR, or some other spectrum-generating experiment.                                           Peak finding\n",
       "2                               https://bio.tools/-CNN                                       -CNN  http://edamontology.org/operation_3222                                                                                                                                                                                                                 Identify putative protein-binding regions in a genome sequence from analysis of Chip-sequencing data or ChIP-on-chip data.                         Protein binding peak detection\n",
       "3                              https://bio.tools/-TRIS                                      -TRIS  http://edamontology.org/operation_2422                         Retrieve an entry (or part of an entry) from a data resource that matches a supplied query. This might include some primary data and annotation. The query is a data identifier or other indexed term. For example, retrieve a sequence record with the specified accession number, or matching supplied keywords.                                        Data extraction\n",
       "4                              https://bio.tools/-TRIS                                      -TRIS  http://edamontology.org/operation_2422                         Retrieve an entry (or part of an entry) from a data resource that matches a supplied query. This might include some primary data and annotation. The query is a data identifier or other indexed term. For example, retrieve a sequence record with the specified accession number, or matching supplied keywords.                                              Retrieval\n",
       "5                              https://bio.tools/-TRIS                                      -TRIS  http://edamontology.org/operation_3198                                                                                                                                                                                                                                                              Align short oligonucleotide sequences (reads) to a larger (genomic) sequence.                              Oligonucleotide alignment\n",
       "6                              https://bio.tools/-TRIS                                      -TRIS  http://edamontology.org/operation_3198                                                                                                                                                                                                                                                              Align short oligonucleotide sequences (reads) to a larger (genomic) sequence.                 Oligonucleotide alignment construction\n",
       "7                              https://bio.tools/-TRIS                                      -TRIS  http://edamontology.org/operation_3198                                                                                                                                                                                                                                                              Align short oligonucleotide sequences (reads) to a larger (genomic) sequence.                   Oligonucleotide alignment generation\n",
       "8                              https://bio.tools/-TRIS                                      -TRIS  http://edamontology.org/operation_3198                                                                                                                                                                                                                                                              Align short oligonucleotide sequences (reads) to a larger (genomic) sequence.                                Oligonucleotide mapping\n",
       "9                              https://bio.tools/-TRIS                                      -TRIS  http://edamontology.org/operation_3198                                                                                                                                                                                                                                                              Align short oligonucleotide sequences (reads) to a larger (genomic) sequence.                                         Read alignment\n",
       "10                             https://bio.tools/-TRIS                                      -TRIS  http://edamontology.org/operation_3198                                                                                                                                                                                                                                                              Align short oligonucleotide sequences (reads) to a larger (genomic) sequence.                        Short oligonucleotide alignment\n",
       "11                             https://bio.tools/-TRIS                                      -TRIS  http://edamontology.org/operation_3198                                                                                                                                                                                                                                                              Align short oligonucleotide sequences (reads) to a larger (genomic) sequence.                                   Short read alignment\n",
       "12                             https://bio.tools/-TRIS                                      -TRIS  http://edamontology.org/operation_3198                                                                                                                                                                                                                                                              Align short oligonucleotide sequences (reads) to a larger (genomic) sequence.                                     Short read mapping\n",
       "13                             https://bio.tools/-TRIS                                      -TRIS  http://edamontology.org/operation_3198                                                                                                                                                                                                                                                              Align short oligonucleotide sequences (reads) to a larger (genomic) sequence.                            Short sequence read mapping\n",
       "14                             https://bio.tools/-TRIS                                      -TRIS  http://edamontology.org/operation_3799                                                                                                                                                                                                                                                             Counting and measuring experimentally determined observations into quantities.                                           Quantitation\n",
       "15                       https://bio.tools/1000genomes                                1000Genomes  http://edamontology.org/operation_0306                                                                                                                                                                                                                                                 Process and analyse text (typically scientific literature) to extract information from it.                                      Literature mining\n",
       "16                       https://bio.tools/1000genomes                                1000Genomes  http://edamontology.org/operation_0306                                                                                                                                                                                                                                                 Process and analyse text (typically scientific literature) to extract information from it.                                         Text analytics\n",
       "17                       https://bio.tools/1000genomes                                1000Genomes  http://edamontology.org/operation_0306                                                                                                                                                                                                                                                 Process and analyse text (typically scientific literature) to extract information from it.                                       Text data mining\n",
       "18                       https://bio.tools/1000genomes                                1000Genomes  http://edamontology.org/operation_3431                                                                                                                                                                                                                                                       Deposit some data in a database or some other type of repository or software system.                                        Data deposition\n",
       "19                       https://bio.tools/1000genomes                                1000Genomes  http://edamontology.org/operation_3431                                                                                                                                                                                                                                                       Deposit some data in a database or some other type of repository or software system.                                        Data submission\n",
       "20                       https://bio.tools/1000genomes                                1000Genomes  http://edamontology.org/operation_3431                                                                                                                                                                                                                                                       Deposit some data in a database or some other type of repository or software system.                                    Database deposition\n",
       "21                       https://bio.tools/1000genomes                                1000Genomes  http://edamontology.org/operation_3431                                                                                                                                                                                                                                                       Deposit some data in a database or some other type of repository or software system.                                    Database submission\n",
       "22                       https://bio.tools/1000genomes                                1000Genomes  http://edamontology.org/operation_3431                                                                                                                                                                                                                                                       Deposit some data in a database or some other type of repository or software system.                                             Submission\n",
       "23    https://bio.tools/1000genomes_assembly_converter             1000Genomes assembly converter  http://edamontology.org/operation_2944                                                                                                                            Generate a physical (sequence) map of a DNA sequence showing the physical distance (base pairs) between features or landmarks such as restriction sites, cloned DNA fragments, genes and other genetic markers.                                   Physical cartography\n",
       "24  https://bio.tools/1000genomes_id_history_converter           1000Genomes ID history converter  http://edamontology.org/operation_3282                                                                                                                                                                                                 Map data identifiers to one another for example to establish a link between two biological databases for the purposes of data integration.                                      Accession mapping\n",
       "25  https://bio.tools/1000genomes_id_history_converter           1000Genomes ID history converter  http://edamontology.org/operation_3282                                                                                                                                                                                                 Map data identifiers to one another for example to establish a link between two biological databases for the purposes of data integration.                                     Identifier mapping\n",
       "26               https://bio.tools/1000genomes_vcf2ped                        1000Genomes VCF2PED  http://edamontology.org/operation_0335                                                                                                                                                                                                                                                                                  Reformat a file of data (or equivalent entity in memory).                                 File format conversion\n",
       "27               https://bio.tools/1000genomes_vcf2ped                        1000Genomes VCF2PED  http://edamontology.org/operation_0335                                                                                                                                                                                                                                                                                  Reformat a file of data (or equivalent entity in memory).                                        File formatting\n",
       "28               https://bio.tools/1000genomes_vcf2ped                        1000Genomes VCF2PED  http://edamontology.org/operation_0335                                                                                                                                                                                                                                                                                  Reformat a file of data (or equivalent entity in memory).                                      File reformatting\n",
       "29               https://bio.tools/1000genomes_vcf2ped                        1000Genomes VCF2PED  http://edamontology.org/operation_0335                                                                                                                                                                                                                                                                                  Reformat a file of data (or equivalent entity in memory).                                      Format conversion\n",
       "30               https://bio.tools/1000genomes_vcf2ped                        1000Genomes VCF2PED  http://edamontology.org/operation_0335                                                                                                                                                                                                                                                                                  Reformat a file of data (or equivalent entity in memory).                                           Reformatting\n",
       "31                       https://bio.tools/13Check_RNA                                13Check_RNA  http://edamontology.org/operation_3431                                                                                                                                                                                                                                                       Deposit some data in a database or some other type of repository or software system.                                        Data deposition\n",
       "32                       https://bio.tools/13Check_RNA                                13Check_RNA  http://edamontology.org/operation_3431                                                                                                                                                                                                                                                       Deposit some data in a database or some other type of repository or software system.                                        Data submission\n",
       "33                       https://bio.tools/13Check_RNA                                13Check_RNA  http://edamontology.org/operation_3431                                                                                                                                                                                                                                                       Deposit some data in a database or some other type of repository or software system.                                    Database deposition\n",
       "34                       https://bio.tools/13Check_RNA                                13Check_RNA  http://edamontology.org/operation_3431                                                                                                                                                                                                                                                       Deposit some data in a database or some other type of repository or software system.                                    Database submission\n",
       "35                       https://bio.tools/13Check_RNA                                13Check_RNA  http://edamontology.org/operation_3431                                                                                                                                                                                                                                                       Deposit some data in a database or some other type of repository or software system.                                             Submission\n",
       "36                          https://bio.tools/1433pred                                14-3-3-Pred  http://edamontology.org/operation_2479                                                                                                                                                                                                                                                  Analyse a protein sequence (using methods that are only applicable to protein sequences).                            Sequence analysis (protein)\n",
       "37                          https://bio.tools/1433pred                                14-3-3-Pred  http://edamontology.org/operation_2575                                                                                                                                                                                                                     Identify or predict catalytic residues, active sites or other ligand-binding sites in protein sequences or structures.                         Protein binding site detection\n",
       "38                          https://bio.tools/1433pred                                14-3-3-Pred  http://edamontology.org/operation_2575                                                                                                                                                                                                                     Identify or predict catalytic residues, active sites or other ligand-binding sites in protein sequences or structures.                        Protein binding site prediction\n",
       "39                               https://bio.tools/16S                                        16S  http://edamontology.org/operation_3799                                                                                                                                                                                                                                                             Counting and measuring experimentally determined observations into quantities.                                           Quantitation\n",
       "40                               https://bio.tools/16S                                        16S  http://edamontology.org/operation_3192                                                                                                                                                                                                                                                                                            Cut (remove) the end from a molecular sequence.                                               Trimming\n",
       "41                    https://bio.tools/16s_classifier                             16S classifier  http://edamontology.org/operation_3460                                                                                                                                                                                                                                              Classifiication (typically of molecular sequences) by assignment to some taxonomic hierarchy.                                    Taxonomy assignment\n",
       "42                            https://bio.tools/16spip                                     16sPIP  http://edamontology.org/operation_2454                                                                                                                                                                                                        Detect, predict and identify genes or components of genes in DNA sequences, including promoters, coding regions, splice sites, etc.                                           Gene calling\n",
       "43                            https://bio.tools/16spip                                     16sPIP  http://edamontology.org/operation_2454                                                                                                                                                                                                        Detect, predict and identify genes or components of genes in DNA sequences, including promoters, coding regions, splice sites, etc.                                           Gene finding\n",
       "44                           https://bio.tools/19andme                                    19andMe  http://edamontology.org/operation_3283                                                                                                                                                                                                                                              Process data in such a way that makes it hard to trace to the person which the data concerns.                                     Data anonymisation\n",
       "45                               https://bio.tools/1KP                                        1KP  http://edamontology.org/operation_3891                                                                                                                                                       Compute Essential Dynamics (ED) on a simulation trajectory: an analysis of molecule dynamics using PCA (Principal Component Analysis) applied to the atomic positional fluctuations.                                                    PCA\n",
       "46                               https://bio.tools/1KP                                        1KP  http://edamontology.org/operation_3891                                                                                                                                                       Compute Essential Dynamics (ED) on a simulation trajectory: an analysis of molecule dynamics using PCA (Principal Component Analysis) applied to the atomic positional fluctuations.                                                     ED\n",
       "47                               https://bio.tools/1KP                                        1KP  http://edamontology.org/operation_3891                                                                                                                                                       Compute Essential Dynamics (ED) on a simulation trajectory: an analysis of molecule dynamics using PCA (Principal Component Analysis) applied to the atomic positional fluctuations.                                        Principal modes\n",
       "48                https://bio.tools/1d-dft-geometrical  Geometrical interpretation of the 1-D DFT  http://edamontology.org/operation_0337                                                                                                                                                                                                                                       Visualise, plot or render (graphically) biomolecular data such as molecular sequences or structures.                                     Data visualisation\n",
       "49                https://bio.tools/1d-dft-geometrical  Geometrical interpretation of the 1-D DFT  http://edamontology.org/operation_0337                                                                                                                                                                                                                                       Visualise, plot or render (graphically) biomolecular data such as molecular sequences or structures.                                              Rendering\n",
       "50                            https://bio.tools/2-kupl                                     2-kupl  http://edamontology.org/operation_3192                                                                                                                                                                                                                                                                                            Cut (remove) the end from a molecular sequence.                                               Trimming\n",
       "51                            https://bio.tools/2-kupl                                     2-kupl  http://edamontology.org/operation_0524                                                                                                                                                                                                                                                Sequence assembly by combining fragments without the aid of a reference sequence or genome.                                        De Bruijn graph\n",
       "52                            https://bio.tools/2-kupl                                     2-kupl  http://edamontology.org/operation_0524                                                                                                                                                                                                                                                Sequence assembly by combining fragments without the aid of a reference sequence or genome.                   Sequence assembly (de-novo assembly)\n",
       "53                            https://bio.tools/2-kupl                                     2-kupl  http://edamontology.org/operation_3227                                                          Detect, identify and map mutations, such as single nucleotide polymorphisms, short indels and structural variants, in multiple DNA sequences. Typically the alignment and comparison of the fluorescent traces produced by DNA sequencing hardware, to study genomic alterations.                                        Variant mapping\n",
       "54                              https://bio.tools/2018                                       2018  http://edamontology.org/operation_3192                                                                                                                                                                                                                                                                                            Cut (remove) the end from a molecular sequence.                                               Trimming\n",
       "55               https://bio.tools/2018_YPIC_Challenge                        2018 YPIC Challenge  http://edamontology.org/operation_0417                                                                                                                                                                                                                                                                          Predict post-translation modification sites in protein sequences.                                           PTM analysis\n",
       "56               https://bio.tools/2018_YPIC_Challenge                        2018 YPIC Challenge  http://edamontology.org/operation_0417                                                                                                                                                                                                                                                                          Predict post-translation modification sites in protein sequences.                                         PTM prediction\n",
       "57               https://bio.tools/2018_YPIC_Challenge                        2018 YPIC Challenge  http://edamontology.org/operation_0417                                                                                                                                                                                                                                                                          Predict post-translation modification sites in protein sequences.                                      PTM site analysis\n",
       "58               https://bio.tools/2018_YPIC_Challenge                        2018 YPIC Challenge  http://edamontology.org/operation_0417                                                                                                                                                                                                                                                                          Predict post-translation modification sites in protein sequences.          Post-translation modification site prediction\n",
       "59               https://bio.tools/2018_YPIC_Challenge                        2018 YPIC Challenge  http://edamontology.org/operation_0417                                                                                                                                                                                                                                                                          Predict post-translation modification sites in protein sequences.               Post-translational modification analysis\n",
       "60               https://bio.tools/2018_YPIC_Challenge                        2018 YPIC Challenge  http://edamontology.org/operation_0417                                                                                                                                                                                                                                                                          Predict post-translation modification sites in protein sequences.                                    PTM site prediction\n",
       "61               https://bio.tools/2018_YPIC_Challenge                        2018 YPIC Challenge  http://edamontology.org/operation_0417                                                                                                                                                                                                                                                                          Predict post-translation modification sites in protein sequences.  Protein post-translation modification site prediction\n",
       "62               https://bio.tools/2018_YPIC_Challenge                        2018 YPIC Challenge  http://edamontology.org/operation_0476                                                                                                                                                                                                                                                     Predict tertiary structure of protein sequence(s) without homologs of known structure.                           de novo structure prediction\n",
       "63                        https://bio.tools/2019ncovas                                 2019nCoVAS  http://edamontology.org/operation_0324                                                                                                                                                                                                                                          Analyse an existing phylogenetic tree or trees, typically to detect features or make predictions.                             Phylogenetic tree analysis\n",
       "64                        https://bio.tools/2019ncovas                                 2019nCoVAS  http://edamontology.org/operation_2403                                                                                                                                                                                                                                                                                             Analyse one or more known molecular sequences.                            Sequence analysis (general)\n",
       "65                          https://bio.tools/2020plus                                   2020plus  http://edamontology.org/operation_2454                                                                                                                                                                                                        Detect, predict and identify genes or components of genes in DNA sequences, including promoters, coding regions, splice sites, etc.                                           Gene calling\n",
       "66                          https://bio.tools/2020plus                                   2020plus  http://edamontology.org/operation_2454                                                                                                                                                                                                        Detect, predict and identify genes or components of genes in DNA sequences, including promoters, coding regions, splice sites, etc.                                           Gene finding\n",
       "67                          https://bio.tools/2020plus                                   2020plus  http://edamontology.org/operation_3197                                                                                                                                                                              Analyse a genetic variation, for example to annotate its location, alleles, classification, and effects on individual transcripts predicted for a gene model.                           Genetic variation annotation\n",
       "68                          https://bio.tools/2020plus                                   2020plus  http://edamontology.org/operation_3197                                                                                                                                                                              Analyse a genetic variation, for example to annotate its location, alleles, classification, and effects on individual transcripts predicted for a gene model.                            Sequence variation analysis\n",
       "69                          https://bio.tools/2020plus                                   2020plus  http://edamontology.org/operation_3197                                                                                                                                                                              Analyse a genetic variation, for example to annotate its location, alleles, classification, and effects on individual transcripts predicted for a gene model.                                       Variant analysis\n",
       "70                            https://bio.tools/2DProt                                    2DProts  http://edamontology.org/operation_2486                                                                                                                                                                                                                                                                                  Render a topology diagram of protein secondary structure.                             Topology diagram rendering\n",
       "71                     https://bio.tools/2StrucCompare                              2StrucCompare  http://edamontology.org/operation_0267                                                                                                                                                                                                                                                                                          Predict secondary structure of protein sequences.               Secondary structure prediction (protein)\n",
       "72                     https://bio.tools/2StrucCompare                              2StrucCompare  http://edamontology.org/operation_2488                                                                                                                                                                                                                                                                                                      Compare protein secondary structures.                            Protein secondary structure\n",
       "73                     https://bio.tools/2StrucCompare                              2StrucCompare  http://edamontology.org/operation_2488                                                                                                                                                                                                                                                                                                      Compare protein secondary structures.               Secondary structure comparison (protein)\n",
       "74        https://bio.tools/2d-image-cepstral-analysis                 2D Image cepstral analysis  http://edamontology.org/operation_3860                                                                                                                                                                                                                                                                     Calculate a theoretical mass spectrometry spectra for given sequences.                                    Spectrum prediction\n",
       "75                             https://bio.tools/2d-mh                                      2D-MH  http://edamontology.org/operation_2479                                                                                                                                                                                                                                                  Analyse a protein sequence (using methods that are only applicable to protein sequences).                            Sequence analysis (protein)\n",
       "76                             https://bio.tools/2d-mh                                      2D-MH  http://edamontology.org/operation_0337                                                                                                                                                                                                                                       Visualise, plot or render (graphically) biomolecular data such as molecular sequences or structures.                                     Data visualisation\n",
       "77                             https://bio.tools/2d-mh                                      2D-MH  http://edamontology.org/operation_0337                                                                                                                                                                                                                                       Visualise, plot or render (graphically) biomolecular data such as molecular sequences or structures.                                              Rendering\n",
       "78                           https://bio.tools/2d-page                                    2D-PAGE  http://edamontology.org/operation_0224                                                                                                                                                                                                                                                                  Search or query a data resource and retrieve entries and / or annotation.                                     Database retrieval\n",
       "79                              https://bio.tools/2dSS                                       2dSS  http://edamontology.org/operation_0570                                                                                                                                                                                                                                        Visualise or render molecular 3D structure, for example a high-quality static picture or animation.                                    Structure rendering\n",
       "80                       https://bio.tools/2de-pattern                                2DE-pattern  http://edamontology.org/operation_2421                                                                                                                                                                                                 Search a database (or other data resource) with a supplied query and retrieve entries (or parts of entries) that are similar to the query.                                                 Search\n",
       "81                          https://bio.tools/2dimpute                                   2DImpute  http://edamontology.org/operation_3557                                                                                                                                                                                                                            Replace missing data with substituted values, usually by using some statistical or other mathematical approach.                                        Data imputation\n",
       "82                              https://bio.tools/2dkd                                       2DKD  http://edamontology.org/operation_2422                         Retrieve an entry (or part of an entry) from a data resource that matches a supplied query. This might include some primary data and annotation. The query is a data identifier or other indexed term. For example, retrieve a sequence record with the specified accession number, or matching supplied keywords.                                        Data extraction\n",
       "83                              https://bio.tools/2dkd                                       2DKD  http://edamontology.org/operation_2422                         Retrieve an entry (or part of an entry) from a data resource that matches a supplied query. This might include some primary data and annotation. The query is a data identifier or other indexed term. For example, retrieve a sequence record with the specified accession number, or matching supplied keywords.                                              Retrieval\n",
       "84                              https://bio.tools/2dkd                                       2DKD  http://edamontology.org/operation_2421                                                                                                                                                                                                 Search a database (or other data resource) with a supplied query and retrieve entries (or parts of entries) that are similar to the query.                                                 Search\n",
       "85                           https://bio.tools/2kplus2                                    2kplus2  http://edamontology.org/operation_2421                                                                                                                                                                                                 Search a database (or other data resource) with a supplied query and retrieve entries (or parts of entries) that are similar to the query.                                                 Search\n",
       "86                           https://bio.tools/2kplus2                                    2kplus2  http://edamontology.org/operation_0484  Find single nucleotide polymorphisms (SNPs) - single nucleotide change in base positions - between sequences. Typically done for sequences from a high-throughput sequencing experiment that differ from a reference genome and which might, especially by reference to population frequency or functional data, indicate a polymorphism.                                            SNP calling\n",
       "87                           https://bio.tools/2kplus2                                    2kplus2  http://edamontology.org/operation_0484  Find single nucleotide polymorphisms (SNPs) - single nucleotide change in base positions - between sequences. Typically done for sequences from a high-throughput sequencing experiment that differ from a reference genome and which might, especially by reference to population frequency or functional data, indicate a polymorphism.                                          SNP discovery\n",
       "88                           https://bio.tools/2kplus2                                    2kplus2  http://edamontology.org/operation_0484  Find single nucleotide polymorphisms (SNPs) - single nucleotide change in base positions - between sequences. Typically done for sequences from a high-throughput sequencing experiment that differ from a reference genome and which might, especially by reference to population frequency or functional data, indicate a polymorphism.               Single nucleotide polymorphism detection\n",
       "89                       https://bio.tools/2lpirnapred                                2lpiRNApred  http://edamontology.org/operation_3936                                                                                                                                                                                                       A dimensionality reduction process that selects a subset of relevant features (variables, predictors) for use in model construction.                                    Attribute selection\n",
       "90                       https://bio.tools/2lpirnapred                                2lpiRNApred  http://edamontology.org/operation_3936                                                                                                                                                                                                       A dimensionality reduction process that selects a subset of relevant features (variables, predictors) for use in model construction.                                     Variable selection\n",
       "91                       https://bio.tools/2lpirnapred                                2lpiRNApred  http://edamontology.org/operation_3936                                                                                                                                                                                                       A dimensionality reduction process that selects a subset of relevant features (variables, predictors) for use in model construction.                              Variable subset selection\n",
       "92                       https://bio.tools/2lpirnapred                                2lpiRNApred  http://edamontology.org/operation_3937                                                                                                                              A dimensionality reduction process which builds (ideally) informative and non-redundant values (features) from an initial set of measured data, to aid subsequent generalization, learning or interpretation.                                     Feature projection\n",
       "93                        https://bio.tools/2passtools                                 2passtools  http://edamontology.org/operation_3198                                                                                                                                                                                                                                                              Align short oligonucleotide sequences (reads) to a larger (genomic) sequence.                              Oligonucleotide alignment\n",
       "94                        https://bio.tools/2passtools                                 2passtools  http://edamontology.org/operation_3198                                                                                                                                                                                                                                                              Align short oligonucleotide sequences (reads) to a larger (genomic) sequence.                 Oligonucleotide alignment construction\n",
       "95                        https://bio.tools/2passtools                                 2passtools  http://edamontology.org/operation_3198                                                                                                                                                                                                                                                              Align short oligonucleotide sequences (reads) to a larger (genomic) sequence.                   Oligonucleotide alignment generation\n",
       "96                        https://bio.tools/2passtools                                 2passtools  http://edamontology.org/operation_3198                                                                                                                                                                                                                                                              Align short oligonucleotide sequences (reads) to a larger (genomic) sequence.                                Oligonucleotide mapping\n",
       "97                        https://bio.tools/2passtools                                 2passtools  http://edamontology.org/operation_3198                                                                                                                                                                                                                                                              Align short oligonucleotide sequences (reads) to a larger (genomic) sequence.                                         Read alignment\n",
       "98                        https://bio.tools/2passtools                                 2passtools  http://edamontology.org/operation_3198                                                                                                                                                                                                                                                              Align short oligonucleotide sequences (reads) to a larger (genomic) sequence.                        Short oligonucleotide alignment\n",
       "99                        https://bio.tools/2passtools                                 2passtools  http://edamontology.org/operation_3198                                                                                                                                                                                                                                                              Align short oligonucleotide sequences (reads) to a larger (genomic) sequence.                                   Short read alignment"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "q= \"\"\"\n",
    "SELECT  * WHERE {\n",
    "?biotools_id rdf:type <http://schema.org/SoftwareApplication> ;\n",
    "   <http://schema.org/name> ?name ; \n",
    "   <http://schema.org/featureList> ?feature .\n",
    "\n",
    "?feature  <http://www.geneontology.org/formats/oboInOwl#hasDefinition> ?def ; \n",
    "         <http://www.geneontology.org/formats/oboInOwl#hasExactSynonym> ?syn .\n",
    "} limit 100\n",
    "\"\"\"\n",
    "print(f\"nb data/operation with has_topic property: {len(remoteQuery(query=q, endpoint=ep_biotools))}\")\n",
    "remoteQuery(query=q, endpoint=ep_biotools)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6aebad53-ce31-48d1-be3c-7c88e7dcec88",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
